### PR TITLE
implemented additional bibliography entry start nodes

### PR DESCRIPTION
**Pull request by eddking**

### CODE REVIEW

File: src/main/java/com/maxdemarzi/shortest/Service.java [hunk 1]
```diff
@@ -325,6 +385,13 @@ private void streamShortestPathsUsingHandwrittenBFS(String centerEmail, List<Str
                         }
                     }
 
+                    if (level == 1) {
+                        // Pretend there are length 1 paths to the bibliograpahy entries
+                        for (Long bibId : bibliographyNodeIds) {
+                            pathsToNextLevel.putIfAbsent(bibId, Integer.valueOf(1));
```
Comment (djudd) [inline src/main/java/com/maxdemarzi/shortest/Service.java]: you don't actually want to put an `Integer` - this is a special collection type that maps primitives to primitives

File: src/main/java/com/maxdemarzi/shortest/Validators.java [hunk 2]
```diff
@@ -22,6 +22,10 @@ public static HashMap getValidQueryInput(String body) throws IOException {
         if (!input.containsKey("center_email")) {
             throw Exceptions.missingCenterEmailParameter;
         }
+        // Make sure it has a bibliography_entries parameter
+        if (!input.containsKey("bibliography_entries")) {
```
Comment (djudd) [inline src/main/java/com/maxdemarzi/shortest/Validators.java:26]: i guess this means we expect downtime when we deploy this, until modify the Sidekiq job and email service to send along bibliography entry IDs? i guess that's ok if we quickly modify the email service to send an empty array, as a placeholder

File: src/main/java/com/maxdemarzi/shortest/Service.java (hunk 1 above)
Comment (eddking) [inline src/main/java/com/maxdemarzi/shortest/Service.java]: they'll get auto-unboxed, but yeah i'll change that

File: src/main/java/com/maxdemarzi/shortest/Validators.java (hunk 2 above)
Comment (eddking) [inline src/main/java/com/maxdemarzi/shortest/Validators.java:26]: yeah, i was planning to deploy a new version of the emailservice first, we shouldnt need any down time

Comment (djudd) [inline src/main/java/com/maxdemarzi/shortest/Validators.java:26]: ok

**Comment by djudd** [comment]
:+1:

**Pull request merged**